packages/connect-form/src/components/advanced-options-tabs/ssh-tunnel-tab/proxy-and-ssh-tunnel-tab.tsx

```diff
@@ -0,0 +1,205 @@
+    searchParams.get('proxyPassword');
+
+  if (isUsingProxy) {
+    return 'socks';
+  }
+
+  if (
+    !connectionOptions ||
+    !connectionOptions?.sshTunnel ||
+    !Object.values(connectionOptions.sshTunnel).find(Boolean) // If the whole object values are empty
+  ) {
+    return 'none';
+  }
+
+  const isUsingIdentity =
+    connectionOptions.sshTunnel.identityKeyFile ||
+    connectionOptions.sshTunnel.identityKeyPassphrase;
+  if (isUsingIdentity) {
+    return 'identity';
+  }
+
+  return 'password';
+};
+
+function ProxyAndSshTunnelTab({
+  connectionOptions,
+  updateConnectionFormField,
+  errors,
+  connectionStringUrl,
+}: {
+  errors: ConnectionFormError[];
+  connectionStringUrl: ConnectionStringUrl;
+  updateConnectionFormField: UpdateConnectionFormField;
+  connectionOptions?: ConnectionOptions;
+}): React.ReactElement {
+  const selectedTunnelType: TunnelType = getSelectedTunnelType(
+    connectionStringUrl,
+    connectionOptions
+  );
+
+  const selectedOptionIndex =
+    options.findIndex((x) => x.type === selectedTunnelType) ?? 0;
+  const [selectedOption, setSelectedOption] = useState(
+    options[selectedOptionIndex]
+  );
+
+  const handleOptionChanged = useCallback(
+    (oldType: TunnelType, newType: TunnelType) => {
+      let type: 'remove-proxy-options' | 'remove-ssh-options';
+      switch (newType) {
+        case 'socks':
+          type = 'remove-ssh-options';
+          break;
+        case 'identity':
+        case 'password':
+          type = 'remove-proxy-options';
+          break;
+        default:
+          type =
+            oldType === 'socks' ? 'remove-proxy-options' : 'remove-ssh-options';
+          break;
+      }
+      updateConnectionFormField({ type });
+    },
+    [updateConnectionFormField]
+  );
+
+  const optionSelected = useCallback(
+    (event: ChangeEvent<HTMLInputElement>) => {
+      event.preventDefault();
+      const item = options.find(({ id }) => id === event.target.value);
+      if (item) {
+        handleOptionChanged(selectedOption.type, item.type);
+        setSelectedOption(item);
+      }
+    },
+    [selectedOption, handleOptionChanged]
+  );
+
+  const SSLOptionContent = selectedOption.component;
+
+  return (
+    <div className={containerStyles}>
+      <Label htmlFor="ssh-options-radio-box-group">
+        SSH Tunnel/Proxy Method
+      </Label>
+      <RadioBoxGroup
+        onChange={optionSelected}
+        className="radio-box-group-style"
+      >
+        {options.map(({ title, id, type }) => {
+          return (
+            <RadioBox
+              data-testid={`${type}-tab-button`}
+              checked={selectedOption.id === id}
+              value={id}
+              key={id}
+            >
+              {title}
+            </RadioBox>
+          );
+        })}
+      </RadioBoxGroup>
+      {connectionOptions && (
+        <div
+          className={contentStyles}
+          data-testid={`${selectedOption.type}-tab-content`}
```

Comment:
My understanding is that `data-testid` values are global, so maybe that’s a good reason to make the values of `TunnelType` a bit less ambiguous as well, e.g. `TunnelType = 'none' | 'ssh-password' | 'ssh-identity' | 'socks'`? Feel free to ignore though, I know that this might involve updating a bunch of code without a strong need

Reply:
well, i ended up implementing this change.